### PR TITLE
Updated node template for when user returns nil

### LINKED ISSUE #89: Update node template for when user returns nil
## Expected Behaviour

Nil should act as a valid return value and just mean that we print nothing.

## Current Behaviour

This causes an exception (i.e. if the user sends "nil" to the callback):

`process.stdout.write(undefined)`

https://github.com/openfaas/templates/blob/master/template/node/index.js#L17

## Possible Solution

Add a truthy check before entering the code block.

## Steps to Reproduce (for bugs)

1. `faas-cli new --lang node test`
2. Edit `test/handler.js`: `module.exports = (context, callback) => { callback(undefined, undefined); }`
3. `faas-cli up --skip-push`
4. Invoke the function. You'll see the error, when fixed you'll just see nothing come back but a 200 status.

## Context

People sometimes see this and are confused about the error. (seen on Slack)

### PR DESCRIPTION
**Pull request by dergoegge**
## Motivation and Context
This fixes the issue descripted in #89.

## Which issue(s) this PR fixes 
Fixes #89 

## How Has This Been Tested?
1. I created a new funtion via: ```faas-cli new --lang node handler-test```
2. I modified ```handler-test/handler.js```: ```module.exports = (context, callback) => { callback(undefined, undefined); }```
3. Invoked the function. ```echo hello | faas-cli invoke handler-test```

With the current version of the template the function threw this error:
```
TypeError: Invalid data, chunk must be a string or buffer, not undefined
    at Socket.write (net.js:701:11)
    at handler (/home/app/index.js:18:28)
    at module.exports (/home/app/function/handler.js:4:5)
    at getStdin.then.val (/home/app/index.js:11:5)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```
With my changes it returned nothing.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
I think the whole block should be ignored by the truthy check. Also it'd great to see the results of your testing before/after. (See the contribution guide)

Thanks,

Alex

**Comment by dergoegge** [comment]
@alexellis i updated my initial comment with testing info!

File: template/node/index.js [hunk 1]
```diff
@@ -23,9 +28,9 @@ getStdin().then(val => {
 });
 
 const isArray = (a) => {
-    return (!!a) && (a.constructor === Array);
```
Comment (alexellis) [inline template/node/index.js]: Curious why you made this change? cc @padiazg

**Comment by alexellis** [comment]
To get the merge:

* revert the unrelated refactoring
* squash the two commits into one (we don't want the first erroneous commit anyway)
* show evidence of this working through the OpenFaaS CLI

Start here if you're new to OpenFaaS: https://blog.alexellis.io/quickstart-openfaas-cli/

This should take you about 5-10 minutes (I'm guessing)

cc @padiazg who raised a similar PR.

File: template/node/index.js (hunk 1 above)
Comment (dergoegge) [reply template/node/index.js]: These functions are being called after my newly added falsy check so i thought that that the !!a is redundant.
But i guess its not related to the actual issue so i should not have removed them.

Comment (alexellis) [reply template/node/index.js]: Thank you

**Comment by dergoegge** [comment]
# Without truthy check
screenshot 2018-10-14 at 11 19 26

# With truthy check
screenshot 2018-10-14 at 11 19 35

**Pull request merged**